openassessment/xblock/grade_mixin.py

```diff
@@ -584,3 +587,94 @@ def _assessment_grade_context(self, assessment):
                     part['option']['label'] = option_labels.get(option_label_key, part['option']['name'])
 
         return assessment
+
+    def _get_score_explanation(self, assessment_steps, status, score):
+        """
+        Return a string which explains how grade is calculated for the ORA assessment
+        based on assessment_steps.
+        Args:
+            assessment_steps (list): List of assessment_steps for some ORA.
+            status (str): String mentioning status of ORA whether it is complete, cancelled, waiting on assessment etc
+            score (dict): Dict containing points_possible, points_earned and its related attributes
+        Returns:
+            str
+        """
+
+        is_self_assessment = "self-assessment" in assessment_steps
+        is_peer_assessment = "peer-assessment" in assessment_steps
+        is_staff_assessment = "staff-assessment" in assessment_steps
+
+        # Assessment Score priorities in descending order
+        assessment_score_priority = AssessmentWorkflow().ASSESSMENT_SCORE_PRIORITY[::-1]
```

Comment:
We can't directly import the `AssessmentWorkflow` from models to inside the XBlock since Django components are not parts of the XBlock API.
I'm not sure how to approach this currently, but the current solution is not acceptable.
The correct way here would be to modify the `AssessmentWorkflowSerializer` and add a property to the model reflecting where the grade came from, but that's out of the question given the task's budget.

Reply:
For the current implementation, consider this priority:
1. Staff grade
2. Peer grade
3. Self grade

This will simplify a lot your logic down below.
Make sure to add a note in the PR description raising this concern (we're not using the `ASSESSMENT_SCORE_PRIORITY` because there's no API for it yet).
I'll create a follow up task to deal with this.

Reply:
If we were to add this information from the Django side to the XBlock side, I would probably modify `self.get_workflow_info()` (and this would go all the way down to the serializers) to get the actual grading priority for this.